diff --git a/CPP/Windows/FileFind.cpp b/CPP/Windows/FileFind.cpp
index 64075ab..ccc4bbd 100755
--- a/CPP/Windows/FileFind.cpp
+++ b/CPP/Windows/FileFind.cpp
@@ -1312,7 +1312,7 @@ bool CDirEntry::IsDots() const throw()
   /* some systems (like CentOS 7.x on XFS) have (Type == DT_UNKNOWN)
      we can call fstatat() for that case, but we use only (Name) check here */
 
-#if !defined(_AIX) && !defined(__sun)
+#if !defined(_AIX) && !defined(__sun) && !defined(__QNX__)
   if (Type != DT_DIR && Type != DT_UNKNOWN)
     return false;
 #endif
@@ -1352,7 +1352,7 @@ bool CEnumerator::NextAny(CDirEntry &fi, bool &found)
 
   fi.iNode = de->d_ino;
   
-#if !defined(_AIX) && !defined(__sun)
+#if !defined(_AIX) && !defined(__sun) && !defined(__QNX__)
   fi.Type = de->d_type;
   /* some systems (like CentOS 7.x on XFS) have (Type == DT_UNKNOWN)
      we can set (Type) from fstatat() in that case.
diff --git a/CPP/Windows/FileFind.h b/CPP/Windows/FileFind.h
index f68673a..af902f9 100755
--- a/CPP/Windows/FileFind.h
+++ b/CPP/Windows/FileFind.h
@@ -277,13 +277,13 @@ typedef CFileInfo CDirEntry;
 struct CDirEntry
 {
   ino_t iNode;
-#if !defined(_AIX) && !defined(__sun)
+#if !defined(_AIX) && !defined(__sun) && !defined(__QNX__)
   Byte Type;
 #endif
   FString Name;
 
   /*
-#if !defined(_AIX) && !defined(__sun)
+#if !defined(_AIX) && !defined(__sun) && !defined(__QNX__)
   bool IsDir() const
   {
     // (Type == DT_UNKNOWN) on some systems
@@ -310,7 +310,7 @@ public:
   bool Fill_FileInfo(const CDirEntry &de, CFileInfo &fileInfo, bool followLink) const;
   bool DirEntry_IsDir(const CDirEntry &de, bool followLink) const
   {
-#if !defined(_AIX) && !defined(__sun)
+#if !defined(_AIX) && !defined(__sun) && !defined(__QNX__)
     if (de.Type == DT_DIR)
       return true;
     if (de.Type != DT_UNKNOWN)
diff --git a/CPP/Windows/System.cpp b/CPP/Windows/System.cpp
index 4745785..60fd914 100755
--- a/CPP/Windows/System.cpp
+++ b/CPP/Windows/System.cpp
@@ -8,6 +8,9 @@
 #if defined(__APPLE__) || defined(__DragonFly__) || \
     defined(BSD) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #include <sys/sysctl.h>
+#elif defined(__QNX__)
+#include <unistd.h>
+#include <sys/conf.h>
 #else
 #include <sys/sysinfo.h>
 #endif
@@ -338,6 +341,13 @@ bool GetRamSize(size_t &size)
     #if defined(_SC_AIX_REALMEM) // AIX
       size64 = (UInt64)sysconf(_SC_AIX_REALMEM) * 1024;
     #endif
+  #elif defined(__QNX__)
+    long pages = sysconf(_SC_PHYS_PAGES);
+    long page_size = sysconf(_SC_PAGESIZE);
+    if (pages == -1 || page_size == -1)
+      size64 = 0;
+    else 
+      size64 = (uint64_t)pages * (uint64_t)page_size;
   #elif 0 || defined(__sun)
     #if defined(_SC_PHYS_PAGES) && defined(_SC_PAGESIZE)
     // FreeBSD, Linux, OpenBSD, and Solaris.
